### ISSUE TITLE
multiply XSS vulnerabilities on contact pages 

### ISSUE THREAD

**Issue by ailyinov** [sulu/sulu#3222]
| Q | A
| --- | ---
| Bug? | yes
| New Feature? | no
| Sulu Version | 1.4.5

#### Actual Behavior

On edit contact details page (`admin/#contacts/contacts/edit:1/details`) add new **position** with value `<script>alert('hi xss!');</script>` and you've got an actual alert.
The same behaviour have fields **title**, **first name**, **last name** .
Also contacts list (`admin/#contacts/contacts`), `admin/#contacts/contacts/edit:1/permissions` and  `admin/#contacts/contacts/edit:1/documents` affected by first name and last name.

**Comment by danrot** [comment]
This is actually a duplicate of https://github.com/sulu/sulu-standard/issues/477, but I close the other issue, because this one is in the correct repository.

**Comment by ailyinov** [comment]
I believe the issue actually comes from [here](https://github.com/massiveart/husky/blob/develop/husky_components/datagrid/decorators/table-view.js#L777) (which is means all components created with datagrid are vulnerable) as far as I understood this calls [jquery `append` method](https://github.com/massiveart/husky/blob/ba38be0e2593cdb11b1179ad35c3def194b269a9/husky_extensions/jquery.js#L78), somehow even if I have escaped data on serverside (`htmlspecialchars`) this xss was still works.
For custom component user could fix this by passing content filter like
```
contentFilters: {
    'title': function(title) {
        return $('<p/>').text(title).html();
     }
}
```
I would suggest make render data as text default behaviour and add another out-of-box filter like `rowHtml` if user want  render some html by purpose.

**Comment by danrot** [comment]
@chirimoya Should we just use something like http://stackoverflow.com/a/4835406/1292378 for the datagrid, the title in the tabs and when collapsing the blocks?

**Comment by danrot** [comment]
The answer in http://stackoverflow.com/questions/24816/escaping-html-strings-with-jquery seems to be a better choice.

**Comment by danrot** [comment]
Closed in #3245
